src/units/attack_type.cpp

```diff
@@ -146,18 +144,24 @@ static bool matches_simple_filter(const attack_type & attack, const config & fil
 	if ( !filter_name.empty() && std::find(filter_name.begin(), filter_name.end(), attack.id()) == filter_name.end() )
 		return false;
 
-	if ( !filter_type.empty() && std::find(filter_type.begin(), filter_type.end(), attack.type()) == filter_type.end() )
-		return false;
-
-	if(!filter_alternative_type.empty() || !filter_replacement_type.empty()){
-		std::pair<std::string, std::string> damage_type = attack.damage_type();
-		if(!filter_replacement_type.empty()){
-			if ( std::find(filter_replacement_type.begin(), filter_replacement_type.end(), damage_type.first) == filter_replacement_type.end() ){
+	if (!filter_type.empty()){
+		if(can_loop){
+			if (std::find(filter_type.begin(), filter_type.end(), attack.type()) == filter_type.end() ){
+				return false;
+			}
+		} else {
+			std::pair<std::string, std::string> damage_type = attack.damage_type();
+			if ((std::find(filter_type.begin(), filter_type.end(), damage_type.first) == filter_type.end()) && (std::find(filter_type.begin(), filter_type.end(), damage_type.second) == filter_type.end())){
 				return false;
 			}
 		}
-		if(!filter_alternative_type.empty()){
-			if ( std::find(filter_alternative_type.begin(), filter_alternative_type.end(), damage_type.second) == filter_alternative_type.end() ){
+	}
+
+	if(can_loop){
+		const std::vector<std::string> filter_modified_type = utils::split(filter["type_changed_by_special"]);
```

Comment:
What is `type_changed_by_special`?

Reply:
in case or someone would condition activity of special with real type(damage modofication incluse), type_changed_by_special replace type in this role. But if you another name to suggest, it's fine.

Reply:
I still don't quite understand. What is this for? Maybe an example would make it clearer.